### ISSUE TITLE
Support intent definition files on iOS

### ISSUE THREAD

**Issue by mpalczew** [facebook/buck#2221]

**Comment by mpalczew** [comment]
Currently buck has no support for intent definition files.  Xcode does two things with these.
 1. it autogenerates code. 
2. it compiles the intentdefinition and includes it in the final ipa.

**Comment by makadaw** [comment]
I don't think it compiles intentdefinition, just add as a resource into intent and an app target, but yes, I can't also use BUCK for autogenerations step.

**Comment by mpalczew** [comment]
I should be more specific. It calls intentbuilderc to turn the intent definition into a resource. 

I think part 2 is the easier step. I think I can do that with just a genrule and then include it as a apple_resource. 

For part 1, it seems Xcode does something completely internal(please tell me I'm wrong here).
